modules/virtual_network_gateway/variables.tf

```diff
@@ -136,59 +129,95 @@ variable "ip_configuration" {
       subnet_id        = "ID_for_subnet_GatewaySubnet"
     }
   ]
-
+  ```
   EOF
-  type        = list(any)
+  default     = []
+  nullable    = false
+  type = list(object({
+    name                          = optional(string, "vnetGatewayConfig")
+    create_public_ip              = bool
+    public_ip_name                = optional(string)
+    private_ip_address_allocation = optional(string, "Dynamic")
+    public_ip_standard_sku        = optional(bool, false)
```

Comment:
do we even consider doing basic?

Reply:
I propose to leave that flexibility to choose e.g. for the lab do we need standard ?